spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/quartz/TracingJobListener.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.instrument.quartz;
+
+import brave.Span;
+import brave.Tracer.SpanInScope;
+import brave.Tracing;
+import brave.propagation.Propagation.Getter;
+import brave.propagation.TraceContextOrSamplingFlags;
+import org.quartz.JobDataMap;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.quartz.JobListener;
+import org.quartz.Trigger;
+import org.quartz.Trigger.CompletedExecutionInstruction;
+import org.quartz.TriggerListener;
+
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * {@link org.quartz.JobListener JobListener} that will wrap a span around quartz jobs
+ * when they start and finish.
+ *
+ * @author Branden Cash
+ * @since 2.2.0
+ */
+public class TracingJobListener implements JobListener, TriggerListener {
+
+	static final String TRIGGER_TAG_KEY = "quartz.trigger";
+
+	static final String CONTEXT_SPAN_KEY = "X-TracingSpan";
+
+	static final String CONTEXT_SPAN_IN_SCOPE_KEY = "X-TracingSpanInScope";
+
+	private static final Getter<JobDataMap, String> GETTER = (carrier, key) -> {
+		if (carrier != null) {
```

Comment:
carrier should never be null, please instead don't pass null to injector or extractor